Libraries/LibWeb/IndexedDB/Internal/Algorithms.cpp

```diff
@@ -864,4 +866,119 @@ ErrorOr<GC::Ref<Key>> convert_a_value_to_a_multi_entry_key(JS::Realm& realm, JS:
     return convert_a_value_to_a_key(realm, value);
 }
 
+// https://w3c.github.io/IndexedDB/#evaluate-a-key-path-on-a-value
+WebIDL::ExceptionOr<ErrorOr<JS::Value>> evaluate_key_path_on_a_value(JS::Realm& realm, JS::Value value, KeyPath key_path)
```

Comment:
you're copying the key_path here, even when it might contain a vector of strings.